### PR TITLE
[StructuredAuthnConfig] add comment for extra keys unique requirement

### PR DESCRIPTION
**Pull request by aramase**
- Add a comment in the API types for the extra keys needed to be unique.

part of https://github.com/kubernetes/kubernetes/issues/121553

/kind cleanup
/sig auth
/triage accepted
/milestone v1.30
/priority important-soon

```release-note
NONE
```

```docs
[KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-auth/3331-structured-authentication-configuration
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go [hunk 1]
```diff
@@ -376,6 +376,7 @@ type ExtraMapping struct {
 	// subdomain as defined by RFC 1123. All characters trailing the first "/" must
 	// be valid HTTP Path characters as defined by RFC 3986.
 	// key must be lowercase.
+	// Required to be unique.
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go:379]: this is already validated, right?

Comment (aramase) [reply staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go:379]: yes, it's validated

https://github.com/kubernetes/kubernetes/blob/1850794626bb995bb754d54be3328c86ee880ba5/staging/src/k8s.io/apiserver/pkg/apis/apiserver/validation/validation.go#L302-L305

**Pull request merged**